### PR TITLE
Basic forum channel support

### PR DESCRIPTION
**Pull request by ryry013**
## Summary

Added basic support for forum channels which are basically guild channels in which users can create threads but cannot send messages in the parent channel. They are registered in the API as channel type `15`. I have tested this patch in my partnered server which has beta access to forums. 

I built the ForumChannel class object by copying CategoryChannel. My experience in the pycord library is limited so I didn't fully write the documentation + `edit()`, `clone()`, etc methods inside the class, but with this patch bots at least fully recognize forum channels and commands work inside them.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/types/channel.py [hunk 1]
```diff
@@ -108,6 +108,10 @@ class StageChannel(_BaseGuildChannel, _StageChannelOptional):
     user_limit: int
 
 
+class ForumChannel(_BaseGuildChannel):
```
Comment (ryry013) [inline discord/types/channel.py:111]: Is `_BaseGuildChannel` fine for this?

File: discord/types/channel.py [hunk 2]
```diff
@@ -108,6 +108,10 @@ class StageChannel(_BaseGuildChannel, _StageChannelOptional):
...
+    type: Literal[15]
+
+
```
Comment (ryry013) [inline discord/types/channel.py:114]: I'm assuming `_BaseGuildChannel` is fine for this.

**Comment by ryry013** [comment]
Forum channel in the channel list: 
image

Main forum channel view:
image

Various tests in the forum channel showing recognition of the channel:
image
image

**Comment by plun1331** [comment]
I feel like there should be some way to get threads that are part of the forum channel, if there isn't one already

Something like `ForumChannel.threads` 

There's an implementation in `TextChannel`:
https://github.com/Pycord-Development/pycord/blob/master/discord/channel.py#L237-L243

I also feel like there are some fields missing, I'm not sure what the whole payload is though.

**Comment by ryry013** [comment]
Adding into a commit now

**Comment by ryry013** [comment]
Working now on writing `create_thread()`. Having trouble because you don't create a message in the parent forum channel to build the thread from, but I don't think they count as private threads either.

Also I realized Discord calls them "posts" in the permissions, it would be a question whether we would call them internally "threads" or "posts" (like, would we call the method `create_post()` or `create_thread()`)

File: discord/channel.py [hunk 3]
```diff
@@ -1808,6 +1810,199 @@ async def edit(self, *, reason=None, **options):
...
+
+    def _update(self, guild: Guild, data: ForumChannelPayload) -> None:
+        self.guild: Guild = guild
+        self.name: str = data["name"]
+        self.category_id: Optional[int] = utils._get_as_snowflake(data, "parent_id")
+        self.nsfw: bool = data.get("nsfw", False)
+        self.position: int = data["position"]
+        self.slowmode_delay: int = data.get('rate_limit_per_user', 0)
+        self.default_auto_archive_duration: ThreadArchiveDuration = data.get('default_auto_archive_duration', 1440)
+        self._type: int = data.get('type', self._type)
```
Comment (Dorukyum) [inline discord/channel.py:1849]: Why does it set `_type` twice?

Comment (ryry013) [reply discord/channel.py:1849]: If you're referring to the copy in `__init__` and the copy in `_update()`, I was basing the code off the `TextChannel()` class: 

https://github.com/Pycord-Development/pycord/blob/master/discord/channel.py#L182

Comment (Dorukyum) [reply discord/channel.py:1849]: I see, thanks for responding

**Comment by Dorukyum** [comment]
Closing in favor of #1249.

**Pull request closed without merging**